ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh

```diff
@@ -199,7 +199,7 @@ cat > packet-setup.yaml <<-EOF
     - cluster_type: "{{ lookup('env', 'CLUSTER_TYPE') }}"
     - packet_project_id: "{{ lookup('file', lookup('env', 'CLUSTER_PROFILE_DIR') + '/packet-project-id') }}"
     - packet_auth_token: "{{ lookup('file', lookup('env', 'CLUSTER_PROFILE_DIR') + '/packet-auth-token') }}"
-
+    - user_data: "{{ lookup('file', lookup('env', 'SHARED_DIR') + '/user-data', errors='ignore') }}"
```

Comment:
Setting the `user-data` via `SHARED_DIR` seems a little bit tricky, since it assumes that a previous step exists and filled it. Wouldn't be easier to fetch it directly from the step env vars (defaulted to empty)?

Reply:
I did not want to have a variable defined in a workflow with a script in it.
Also, having a step gives us more freedom is we want to generate it from a template.

What I can do is to define a new variable that specifies the user-data filename in `SHARED_DIR`. This variable would be empty by default. What do you think?

Reply:
What's the benefit with that last suggestion? It makes it possible for steps to choose the location for such files?
Because I think it's mostly a downside, as it means there's no real convention to where it resides (and either way ci-operator does not support calling steps twice with different parameters, so generalizing it doesn't provide much)

Reply:
It's a way to signal that a file is available to be consumed, and ensure a failure will be risen if it is not there.

The file must reside in `SHARED_DIR` and comply with the limitation of this folder, i.e: [flat hierarchy](https://docs.ci.openshift.org/docs/architecture/step-registry/#sharing-data-between-steps). So at the end, it would be a variable `USER_DATA_FILENAME` set at workflow level with the filename that is wanted (e.g. "user-data"), the step that generates the user data would generate the script in `${SHARED_DIR}/${USER_DATA_FILENAME}`. and the ansible script would look for it in `${SHARED_DIR}/${USER_DATA_FILENAME}`.

This is what I had in mind.

Reply:
ack. so it does make sense to do that and make things more explicit rather than implicit 
sounds good to me

Reply:
I understand that in some cases having it as a step parameter could not be too handy, even though I wouldn't expect in general to see very big user-data contents (if that's the case, and if there's the need to heavily customize the instance, my preference would be to have an Ansible customization step somewhere).
At least, and agree with @osherdp  on that, the definition of the new variable makes more clear the step documentation part for a generic user, so it also sounds reasonable to me.